src/main/java/me/arminb/sara/controller/UserController.java

```diff
@@ -0,0 +1,118 @@
+package me.arminb.sara.controller;
+
+import me.arminb.sara.dao.DataAccessException;
+import me.arminb.sara.entities.User;
+import me.arminb.sara.services.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+
+    @Autowired
+    private UserService userService;
+
+    @RequestMapping(method = RequestMethod.GET)
+    public ResponseEntity<List<User>> findAll() {
+        try {
+            List<User> users = userService.findAll();
+            return new ResponseEntity<List<User>>(users, HttpStatus.OK);
+        }
+        catch (DataAccessException e){
+            return new ResponseEntity<List<User>>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+        catch (Exception e){
```

Comment:
I would be more specific about the exceptions. Catching Exception, RuntimeException or Throwable is not a good practice in general since you may catch more specific exceptions in this catch clause when you are not expecting them. This can lead to improper handling of some important exceptions. In this case, it is not clear why a BAD REQUEST status code should be returned. We only return BAD REQUEST when the request cannot be parsed or comprehended.